addons/test_mail/tests/common.py

```diff
@@ -132,6 +132,16 @@ def sudoAs(self, login):
             self.env = self.env(user=self.uid)
             self.test_record = self.test_record_old
 
+    def formataddr_superuser(self, partner_from):
+        if partner_from._name == 'res.users':
+            partner_from = partner_from.partner_id
+        if partner_from == self.env['res.users'].browse(SUPERUSER_ID).partner_id:
+            company = self.env.user.company_id
+            format_name = '%s (%s)'
+            return formataddr((format_name % (partner_from.name, company.name), company.email)) if company.email else company.catchall
```

Comment:
better call ~~saul~~ `self.env['mail.message'].sudo()._get_default_from()`

Reply:
I wanted to do that, but it would pollute the default_from:
- in the tests, we explicitly pass "author", so I need to put it as argument of default_from just for this case
- in the business code, we enter the default only if the email_from is not set beforehand
Those are two different semantics, which I can implement if judged worth it.
A middle ground could be to have a odoo.tools helper function, but I judged that overkill for a first draft of this PR, given that the fix already adds some crap for a specific case.
I take the opportunity to throw here the ideas that a discussion with @Xavier-Do rose:
- in master version, make the computation of email_from, reply_to and author less spaghetti
This would probably means a (huge) refactor
- my fix could implement or call get_reply_to to determine the email address, i.e. check if any alias would match. It's not a bad idea, but may be overkill for the issue at hand